PopPUNK/lineage_clustering.py

```diff
@@ -0,0 +1,680 @@
+            lineage_clustering = run_lineage_clustering(lineage_clustering,
+                                                        lineage_clustering_information,
+                                                        neighbours,
+                                                        R,
+                                                        lineage_index,
+                                                        seed_isolate,
+                                                        previous_lineage_clustering,
+                                                        null_cluster_value)
+                                                        
+        # increment index for next lineage
+        lineage_index = lineage_index + 1
+
+    # return clustering
+    return lineage_clustering, lineage_seed, neighbours, previous_lineage_clustering
+
+def calculateQueryDistances(dbFuncs, rlist, qfile, kmers, estimated_length,
+                    queryDB, use_mash = False, threads = 1):
+    """Finds edges between queries and items in the reference database,
+    and modifies the network to include them.
+
+    Args:
+        dbFuncs (list)
+            List of backend functions from :func:`~PopPUNK.utils.setupDBFuncs`
+        rlist (list)
+            List of reference names
+        qfile (str)
+            File containing queries
+        kmers (list)
+            List of k-mer sizes
+        estimated_length (int)
+            Estimated length of genome, if not calculated from data
+        queryDB (str)
+            Query database location
+        use_mash (bool)
+            Use the mash backend
+        threads (int)
+            Number of threads to use if new db created
+            (default = 1)
+    Returns:
+        qlist1 (list)
+            Ordered list of queries
+        distMat (numpy.array)
+            Query-query distances
+    """
+    
+    constructDatabase = dbFuncs['constructDatabase']
+    queryDatabase = dbFuncs['queryDatabase']
+    readDBParams = dbFuncs['readDBParams']
+
+    # These are returned
+    qlist1 = None
+    distMat = None
+
+    # Set up query names
+    qList, qSeqs = readRfile(qfile, oneSeq = use_mash)
+    queryFiles = dict(zip(qList, qSeqs))
+    if use_mash == True:
+        rNames = None
+        qNames = qSeqs
+    else:
+        rNames = qList
+        qNames = rNames
+
+    # Calculate all query-query distances too, if updating database
+    qlist1, qlist2, distMat = queryDatabase(rNames = rNames,
+                                            qNames = qNames,
+                                            dbPrefix = queryDB,
+                                            queryPrefix = queryDB,
+                                            klist = kmers,
+                                            self = True,
+                                            number_plot_fits = 0,
+                                            threads=threads)
+
+    return qlist1, distMat
+
+
+def readLineages(clustCSV):
```

Comment:
Think about combining this with readClusters

Reply:
Merged readLineages, readClusters, readExternalClusters into readIsolateTypeFromCsv in utils.py in 14ed2f4.